src/axom/primal/geometry/NumericArray.hpp

```diff
@@ -428,19 +442,19 @@ NumericArray< T, SIZE >::NumericArray(const T* vals, int sz)
   const int nvals = axom::utilities::clampVal(sz, 0, SIZE);
 
   // Copy first nvals coordinates from vals array ( 0 <= nvals <= SIZE )
-  std::copy( vals, vals+nvals, m_components);
+  for (int i = 0; i < nvals; i++) {
+    m_components[i] = vals[i];    
+  }
 
   // Fill any remaining coordinates with zero
-  if ( nvals < SIZE)
-  {
-    std::fill( m_components+nvals, m_components+SIZE, T());
+  for (int j = nvals; j < SIZE; j++) {
+    m_components[j] = T();    
   }
-
 }
 
 //------------------------------------------------------------------------------
 template < typename T,int SIZE >
-inline NumericArray< T,SIZE >&
+inline  AXOM_HOST_DEVICE NumericArray< T,SIZE >&
 NumericArray< T,SIZE >::operator=( const NumericArray< T,SIZE >& rhs )
 {
 
```

Comment:
@bmhan12 -- I am making some small edits as I review this PR.

You replaced all the ``std::fill`` and ``std::copy`` calls above.
Is the ``memcpy`` below ok on the device? Or does it just not get called in any of our code paths?

Reply:
Yes,` memcpy` is ok on device.